docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,898 @@
+
+    script:
+    """
+    echo "Sample metadata: ${id}" > ${id}_stats.txt
+    echo "Replicate: ${replicate}" >> ${id}_stats.txt
+    echo "Type: ${type}" >> ${id}_stats.txt
+    echo "Read 1: ${fastqs[0]}" >> ${id}_stats.txt
+    echo "Read 2: ${fastqs[1]}" >> ${id}_stats.txt
+    echo "File sizes:" >> ${id}_stats.txt
+    echo "Read 1 size: \$(wc -l < ${fastqs[0]} | awk '{print \$1/4}') reads" >> ${id}_stats.txt
+    echo "Read 2 size: \$(wc -l < ${fastqs[1]} | awk '{print \$1/4}') reads" >> ${id}_stats.txt
+    """
+}
+
+workflow {
+```
+
+### 5.2. Implement the process in the workflow
+
+Then implement the process in the workflow:
+
+_Before_:
+
+```groovy title="main.nf" linenums="31"
+    .view()
+}
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="31"
+    ch_samples = ch_fastq.map { id, fastqs ->
+        def (sample, replicate, type, readNum) = id.tokenize('_')
+        [ sample, replicate.replace('rep', ''), type, fastqs ]
+    }
+
+    ANALYZE_READS(ch_samples)
+}
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="ANALYZE_READS Output"
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `./main.nf` [shrivelled_cori] DSL2 - revision: b546a31769
+
+executor >  local (1)
+[b5/110360] process > ANALYZE_READS (sampleA) [100%] 1 of 1 ✔
+```
+
+We should see the following files in the `results/sampleA` directory:
+
+```console title="Results Directory"
+> tree results/sampleA
+results/sampleA
+└── sampleA_stats.txt
+```
+
+The process took our inputs and created a new file with the sample metadata. Based on what you learned in hello-nextflow, what occurred in the working directory?
+
+### 5.3. Include many more samples
+
+Remember Channel.fromPath() accepts a _glob_ as input, which means it can accept any number of files that match the pattern. Therefore if we want to include all the samples we can just modify the input string to include more samples.
+
+_Before_:
+
+```groovy title="main.nf" linenums="26"
+ch_fastq = Channel.fromFilePairs('data/sampleA_rep1_normal_R{1,2}_001.fastq.gz')
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="26"
+ch_fastq = Channel.fromFilePairs('data/*_R{1,2}_001.fastq.gz')
+```
+
+Run the pipeline now and see all the results:
+
+```bash
+nextflow run main.nf
+```
+
+```console title="ANALYZE_READS Multiple Samples"
+ N E X T F L O W   ~  version 24.10.5
+
+Launching `./main.nf` [big_stonebraker] DSL2 - revision: f7f9b8a76c
+
+executor >  local (8)
+[d5/441891] process > ANALYZE_READS (sampleC) [100%] 8 of 8 ✔
+```
+
+Check the results directory now:
+
+```console title="Results Directory"
+> tree results
+results
+├── sampleA
+│   └── sampleA_stats.txt
+├── sampleB
+│   └── sampleB_stats.txt
+└── sampleC
+    └── sampleC_stats.txt
+```
+
+See how we have analyzed all the samples in one go!
+
+Wait, we have a problem. We have 2 replicates for sampleA, but only 1 output file! We are overwriting the output file each time.
+
+### 5.4. Make the published files unique
```

Comment:
I suppose this is to use something from the meta info as example on how it can be used in the workflow. Should we do something similar as in the nf-core module or change the nf-core module (we used branch to change the execution path) ? Just to revisit the same concepts again